service/cloudconfig/cloud_config.go

```diff
@@ -1,43 +1,267 @@
+				FileName:   "/etc/kubernetes/ssl/etcd/server-ca.pem",
+			},
+			keyVaultSecret{
+				SecretName: key.SecretName(certificatetpr.EtcdComponent, certificatetpr.Crt),
+				FileName:   "/etc/kubernetes/ssl/etcd/server-crt.pem",
+			},
+			keyVaultSecret{
+				SecretName: key.SecretName(certificatetpr.EtcdComponent, certificatetpr.Key),
+				FileName:   "/etc/kubernetes/ssl/etcd/server-key.pem",
+			},
+			// Service account.
+			keyVaultSecret{
+				SecretName: key.SecretName(certificatetpr.ServiceAccountComponent, certificatetpr.CA),
+				FileName:   "/etc/kubernetes/ssl/service-account-ca.pem",
+			},
+			keyVaultSecret{
+				SecretName: key.SecretName(certificatetpr.ServiceAccountComponent, certificatetpr.Crt),
+				FileName:   "/etc/kubernetes/ssl/service-account-crt.pem",
+			},
+			keyVaultSecret{
+				SecretName: key.SecretName(certificatetpr.ServiceAccountComponent, certificatetpr.Key),
+				FileName:   "/etc/kubernetes/ssl/service-account-key.pem",
+			},
+		},
+	}
+
+	getSecretsScript, err := me.renderGetSecretsScript(secrets)
+	if err != nil {
+		return k8scloudconfig.FileAsset{}, microerror.Mask(err)
+	}
 
-// Files allows files to be injected into the worker cloudconfig. It is
-// currently empty because the certificates will be stored in Key Vault.
+	return getSecretsScript, nil
+}
+
+// Files allows files to be injected into the worker cloudconfig.
 func (we *WorkerExtension) Files() ([]k8scloudconfig.FileAsset, error) {
-	return nil, nil
+	var newFiles []k8scloudconfig.FileAsset
+
+	getSecretsScript, err := we.getWorkerSecretsScript()
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	newFiles = []k8scloudconfig.FileAsset{
+		getSecretsScript,
+	}
+
+	return newFiles, nil
 }
 
 // Units allows systemd units to be injected into the worker cloudconfig.
-// TODO Add unit for downloading certificates from Key Vault.
 func (we *WorkerExtension) Units() ([]k8scloudconfig.UnitAsset, error) {
-	return nil, nil
+	unitsMeta := []k8scloudconfig.UnitMetadata{
+		{
+			AssetContent: getKeyVaultSecretsUnit,
+			Name:         "get-keyvault-secrets.service",
+			Enable:       true,
+			Command:      "start",
+		},
+	}
+
+	units, err := we.renderUnits(unitsMeta)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	return units, nil
 }
 
 // VerbatimSections allows sections to be embedded in the worker cloudconfig.
 func (we *WorkerExtension) VerbatimSections() []k8scloudconfig.VerbatimSection {
 	return nil
 }
+
+// getWorkerSecretsScript returns the script for downloading the worker TLS
+// certificates from Key Vault on startup.
+func (we *WorkerExtension) getWorkerSecretsScript() (k8scloudconfig.FileAsset, error) {
+	secrets := keyVaultSecrets{
+		VaultName: key.KeyVaultName(we.CustomObject),
+		Secrets: []keyVaultSecret{
+			// Calico client.
+			keyVaultSecret{
+				SecretName: key.SecretName(certificatetpr.CalicoComponent, certificatetpr.CA),
+				FileName:   "/etc/kubernetes/ssl/calico/client-ca.pem",
```

Comment:
We may want to extract those as consts in kyes package to improve visibility of files we create.

Reply:
We don't do that in AWS or KVM but it would be a nice improvement. The keys package is a good location I think.

Reply:
Uncertain. File paths are no keys.

Reply:
I think they are. At least "key candidates". For example if we start to configure cert path in the TPO then we would be clear this is a key. I think it's a good idea to store potential "key candidates" in the package as well. This gives nice overview of where we store and how we name things.